src/v0/sources/shopify/shopify.util.test.js

```diff
@@ -0,0 +1,191 @@
+            const input = {
+                randomKey: 'randomValue',
+            };
+            const expectedOutput = {
+                error: 'Query_parameters is missing',
+            };
+            try {
+                getShopifyTopic(input);
+            } catch (error) {
+                expect(error.message).toEqual(expectedOutput.error);
+            }
+        });
+    });
+
+    describe('set AnonymousId without using Redis Test Cases', () => {
+        it('Properties containing cartToken', () => {
+            const input = {
+                event: 'Order Updated',
+                properties: {
+                    cart_token: '123',
+                }
+            };
+            const expectedOutput = 'a665a45920422f9d417e4867efdc4fb8a04a';
+            const output = getAnonymousId(input);
+            expect(output).toEqual(expectedOutput);
+        });
+
+        it('Properties contain id for cart event', () => {
+            const input = {
+                event: 'Cart Update',
+                properties: {
+                    id: '123',
+                }
+            };
+
+            const expectedOutput = 'a665a45920422f9d417e4867efdc4fb8a04a';
+            const output = getAnonymousId(input);
+            expect(output).toEqual(expectedOutput);
+        });
+
+        it('Customer event -> random AnonymousId', () => {
+            const input = {
+                event: 'Customer Enabled',
+                properties: {
+                    cart_token: '123',
+                }
+            };
+            const output = getAnonymousId(input);
+            expect(output).toEqual(output); // since it will be random
+        });
+
+        it('Order Delete -> No anonymousId is there', () => {
+            const input = {
+                event: 'Order Deleted',
+                properties: {
+                    cart_token: '123',
+                }
+            };
+            const expectedOutput = null;
+            const output = getAnonymousId(input);
+            expect(output).toEqual(expectedOutput);
+        });
+    });
+    describe('set AnonymousId with Redis Test Cases', () => {
+        it('Properties containing cartToken', async () => {
+            const input = {
+                event: 'Order Paid',
+                properties: {
+                    cart_token: 'shopify_test2',
+                }
+            };
+            const expectedOutput = 'anon_shopify_test2'
+            const output = await getAnonymousIdFromDb(input);
+            expect(output).toEqual(expectedOutput);
+        });
+
+        it('Properties contain id for cart event', async () => {
+            const input = {
+                event: 'Cart Update',
+                properties: {
+                    id: 'shopify_test2',
+                }
+            };
+
+            const expectedOutput = 'anon_shopify_test2';
+            const output = await getAnonymousIdFromDb(input);
```

Comment:
Add a test case where we are fetching `anonymousId` from Db but its not there in redis

Reply:
Added